### ISSUE TITLE
no-eval false negatives

### ISSUE THREAD

**Issue by cknight** [denoland/deno_lint#466]
es-lint flags the following three examples with the no-eval rule which deno lint does not:
```javascript
(0, eval)("var a = 0");

var foo = eval;
foo("var a = 0");

this.eval("var a = 0");
```

**Comment by ross-weir** [comment]
I'll try fix this one 👍
